ai-agent/api/routers/search_offers.py

```
@@ -0,0 +1,44 @@
+)
+
+@router.post("/generate-sheet")
+async def create_information_sheet(profile: UserProfile):
+    """
+    フロントエンドから送られてきたユーザー経歴情報をもとに求人検索をするエンドポイント
+    """
+    try:
+        response = search_personalized_offers(profile)
+        return {"status": "success", "response": response}
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Sheet generation failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)